application/src/main/scala/com/azavea/franklin/tile/CogAssetNode.scala

```diff
@@ -0,0 +1,91 @@
+package com.azavea.franklin.tile
+
+import cats.data.{NonEmptyList => NEL}
+import cats.effect.IO
+import com.azavea.franklin.cache._
+import com.azavea.stac4s.StacItemAsset
+import geotrellis.layer._
+import geotrellis.proj4._
+import geotrellis.raster._
+import geotrellis.raster.geotiff.GeoTiffRasterSource
+import geotrellis.raster.histogram.Histogram
+import geotrellis.raster.resample._
+import scalacache.modes.sync._
+import scalacache.{Sync => _, _}
+
+case class CogAssetNode(asset: StacItemAsset, bands: Seq[Int]) extends TileUtil {
+  private val histoKey = s"histogram - $bands - ${asset.href}"
+  private val tiffKey  = s"tiff - ${asset.href}"
+
+  def getRasterSource: IO[GeoTiffRasterSource] = {
+
+    val infoFromCache = sync.get[SerializableGeotiffInfo](tiffKey)
+    val tiffInfoIO = infoFromCache match {
+      case Some(info) => IO(info)
```

Comment:
I think you probably want an `IO.pure(info)` here, since `IO.apply` has a by name parameter. doubtful it makes much difference per tile, but maybe with a lot of simultaneous tile requests the thunks could matter